src/member.rs

```diff
@@ -111,111 +84,96 @@ impl<H: Hasher> ScheduledTask<H> {
     }
 }
 
-impl<H: Hasher> Ord for ScheduledTask<H> {
+impl<H: Hasher, D: Data, S: Signature> Ord for ScheduledTask<H, D, S> {
     fn cmp(&self, other: &Self) -> Ordering {
         // we want earlier times to come first when used in max-heap, hence the below:
         other.scheduled_time.cmp(&self.scheduled_time)
     }
 }
 
-impl<H: Hasher> PartialOrd for ScheduledTask<H> {
+impl<H: Hasher, D: Data, S: Signature> PartialOrd for ScheduledTask<H, D, S> {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.cmp(other))
     }
 }
 
-/// A representation of a committee member responsible for establishing the consensus.
-///
-/// It depends on the following objects (for more detailed description of the above obejcts, see their references):
-/// - [`Hasher`] - an abstraction for creating identifiers for units, alerts, and other internal objects,
-/// - [`DataIO`] - an abstraction for a component that outputs data items and allows to input ordered data items,
-/// - [`MultiKeychain`] - an abstraction for digitally signing arbitrary data and verifying signatures,
-/// - [`Network`] - an abstraction for a network connecting the committee members,
-/// - [`SpawnHandle`] - an abstraction for an executor of asynchronous tasks.
-///
-/// For a detailed description of the consensus implemented in Member see
-/// [docs for devs](https://cardinal-cryptography.github.io/AlephBFT/index.html)
-/// or the [original paper](https://arxiv.org/abs/1908.05156).
-pub struct Member<'a, H, D, DP, MK, SH>
+struct Member<H, D, S>
 where
     H: Hasher,
     D: Data,
-    DP: DataIO<D>,
-    MK: MultiKeychain,
-    SH: SpawnHandle,
+    S: Signature,
 {
     config: Config,
-    tx_consensus: Option<Sender<NotificationIn<H>>>,
-    data_io: DP,
-    keybox: &'a MK,
-    store: UnitStore<'a, H, D, MK>,
-    task_queue: BinaryHeap<ScheduledTask<H>>,
-    requested_coords: HashSet<UnitCoord>,
-    threshold: NodeCount,
+    task_queue: BinaryHeap<ScheduledTask<H, D, S>>,
+    not_resolved_parents: HashSet<H::Hash>,
+    not_resolved_coords: HashSet<UnitCoord>,
     n_members: NodeCount,
-    unit_messages_for_network: Option<Sender<(UnitMessage<H, D, MK::Signature>, Recipient)>>,
-    alerts_for_alerter: Option<Sender<Alert<H, D, MK::Signature>>>,
-    spawn_handle: SH,
+    unit_messages_for_network: Sender<(UnitMessage<H, D, S>, Recipient)>,
+    unit_messages_from_network: Receiver<UnitMessage<H, D, S>>,
+    messages_from_network: Sender<RunwayNotificationIn<H, D, S>>,
+    messages_for_network: Receiver<RunwayNotificationOut<H, D, S>>,
```

Comment:
Shouldn't these two be named `notifications_for_runway` and `notifications_from_runway` respecitively?

Reply:
renamed